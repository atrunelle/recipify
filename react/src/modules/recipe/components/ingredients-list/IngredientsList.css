.ingredients-list__list {
  padding: 16px 0 16px 0; }
