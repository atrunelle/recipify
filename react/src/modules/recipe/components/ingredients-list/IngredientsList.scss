.ingredients-list {
  &__list {
    padding: 16px 0 16px 0;
  }
}