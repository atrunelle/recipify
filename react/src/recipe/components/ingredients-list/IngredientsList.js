import React, { Component } from 'react';
// import './ingredientsList.css';


class IngredientsList extends Component {

    render() {
        if (!this.props.ingredients.healthLabels) return (<div></div>);
        const healthLabels = this.props.ingredients.healthLabels.reduce((acc, h) => {
            return `${acc} <br> ${h}`;
        })
        return(
            <div className="results">
                <h2>{this.props.title}</h2>
                <div>{this.props.ingredients.calories}</div>
                <div>{this.props.ingredients.totalWeight}</div>
                <div>{this.props.ingredients.dietLabels}</div>
                <div>{healthLabels}</div>
                {/* {this.props.results.totalDaily} */}
                {/* {this.props.results.totalNutrients} */}
                <div>{this.props.ingredients.cautions}</div>
                <div>{this.props.ingredients.glycemicIndex}</div>
                {/* {this.props.results.ingredients} */}
            </div>
        );
    }
}

export default IngredientsList;