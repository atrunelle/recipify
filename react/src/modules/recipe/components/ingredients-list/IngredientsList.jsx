import React, { Component } from 'react';
import { 
  TextField, Button, Typography, Grid,
} from '@material-ui/core';
import IngredientDetails from '../ingredient-details/IngredientDetails';
import './IngredientsList.css';

class IngredientsList extends Component {
  constructor(props) {
    super(props);

    this.state = {
      recipeName: '',
    };
  }

  onRecipeNameChange = (event) => {
    this.setState({ recipeName : event.target.value });
  }

  save = () => {
    this.props.save(this.state.recipeName);
  }

  render() {
    if (!this.props.ingredients) return false;
  
    const ingredientsList = this.props.ingredients.map((ingredient, index) => 
      <IngredientDetails key={index} index={index} ingredient={ingredient} remove={this.props.remove}/>
    )
    
    return(
      <Grid 
        container
        direction="column">
        <Grid
          container
          direction="row"
          justify="space-between"
          alignItems="center">
          <Typography
            gutterBottom 
            variant="headline"
            component="h2">
            Your recipe ingredients list
          </Typography>          
          <form>
            <TextField
              label="Recipe name"
              value={this.state.recipeName}
              id="recipeNameInput"
              onChange={this.onRecipeNameChange}
            />
          </form>
        </Grid>
        <Grid
          item
          className="ingredients-list__list">
          {ingredientsList}
        </Grid>
        <Grid
          container
          spacing={8}
          justify="flex-end">
          <Grid item>
            <Button
              onClick={this.props.removeAll}
              id="removeAllButton">
              Remove all ingredients
            </Button>
          </Grid>
          <Grid item>
            <Button
              variant="raised"
              color="primary"
              id="saveButton"
              onClick={this.save}>
              Save recipe
            </Button>
          </Grid>
        </Grid>
      </Grid>
    );
  }
}

export default IngredientsList;