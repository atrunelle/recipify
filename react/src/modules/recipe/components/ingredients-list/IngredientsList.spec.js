import React from 'react';
import { shallow } from 'enzyme';
import IngredientsList from './IngredientsList';

describe('Component: IngredientsList', () => {
  it('should not render if no ingredients props is passed', () => {
    const component = shallow(<IngredientsList />);
    expect(component.html()).toMatchSnapshot();
  });

  describe('when ingredients prop is passed', () => {
    let component;
    const mockSave = jest.fn();
    const mockRemoveAll = jest.fn();
    const mockIngredients = [{
      name: 'Carrot',
      nutrients: {
        calories: 100,
        totalWeight: 260,
        totalNutrients: {},
        totalDaily: {},
        dietLabels: [],
        healthLabels: [],
      },
    }];

    beforeEach(() => {
      component = shallow(
        <IngredientsList 
          ingredients={mockIngredients}
          save={mockSave}
          removeAll={mockRemoveAll}/>);      
    });
    
    it('should match snapshot', () => {
      expect(component.html()).toMatchSnapshot();
    });
  });
});