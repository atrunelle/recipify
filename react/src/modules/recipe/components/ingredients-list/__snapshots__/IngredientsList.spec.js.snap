// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: IngredientsList should not render if no ingredients props is passed 1`] = `null`;

exports[`Component: IngredientsList when ingredients prop is passed should match snapshot 1`] = `"<div class=\\"MuiGrid-container-1 MuiGrid-direction-xs-column-4\\"><div class=\\"MuiGrid-container-1 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-space-between-20\\"><h2 class=\\"MuiTypography-root-92 MuiTypography-headline-97 MuiTypography-gutterBottom-109\\">Your recipe ingredients list</h2><form><div class=\\"MuiFormControl-root-116\\"><label class=\\"MuiFormLabel-root-125 MuiInputLabel-root-120 MuiInputLabel-formControl-121 MuiInputLabel-animated-124\\" data-shrink=\\"false\\" for=\\"recipeNameInput\\">Recipe name</label><div class=\\"MuiInput-root-130 MuiInput-formControl-131 MuiInput-underline-134\\"><input type=\\"text\\" aria-invalid=\\"false\\" aria-required=\\"false\\" class=\\"MuiInput-input-138\\" id=\\"recipeNameInput\\" value=\\"\\"/></div></div></form></div><div class=\\"MuiGrid-item-2 ingredients-list__list\\"><div class=\\"MuiPaper-root-146 MuiPaper-elevation1-149 MuiExpansionPanel-root-143 ingredient-details\\"><div tabindex=\\"0\\" class=\\"MuiButtonBase-root-179 MuiExpansionPanelSummary-root-173\\" role=\\"button\\" aria-expanded=\\"false\\"><div class=\\"MuiExpansionPanelSummary-content-177\\"><h3 class=\\"MuiTypography-root-92 MuiTypography-body1-101\\">Carrot</h3><p class=\\"MuiTypography-root-92 MuiTypography-body1-101\\">100cal for 260gr</p><button tabindex=\\"0\\" class=\\"MuiButtonBase-root-179 MuiIconButton-root-182 MuiIconButton-colorSecondary-185 ingredient-details__button\\" type=\\"button\\" aria-label=\\"Delete\\" id=\\"removeButton-0\\"><span class=\\"MuiIconButton-label-187\\"><svg class=\\"MuiSvgIcon-root-188\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><g><path d=\\"M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z\\"></path></g></svg></span><span class=\\"MuiTouchRipple-root-194\\"></span></button></div><div tabindex=\\"-1\\" class=\\"MuiButtonBase-root-179 MuiIconButton-root-182 MuiExpansionPanelSummary-expandIcon-178\\" role=\\"button\\" aria-hidden=\\"true\\"><span class=\\"MuiIconButton-label-187\\"><svg class=\\"MuiSvgIcon-root-188\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><g><path d=\\"M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z\\"></path></g></svg></span><span class=\\"MuiTouchRipple-root-194\\"></span></div></div><div class=\\"MuiCollapse-container-201\\" style=\\"min-height:0px\\" aria-hidden=\\"true\\"><div class=\\"MuiCollapse-wrapper-203\\"><div class=\\"MuiCollapse-wrapperInner-204\\"><div class=\\"MuiExpansionPanelDetails-root-205\\"><div class=\\"MuiGrid-container-1 MuiGrid-spacing-xs-8-22 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-center-18\\"><div class=\\"MuiGrid-container-1 MuiGrid-justify-xs-space-between-20\\"></div><div class=\\"MuiGrid-container-1 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-center-18\\"></div></div></div></div></div></div></div></div><div class=\\"MuiGrid-container-1 MuiGrid-spacing-xs-8-22 MuiGrid-justify-xs-flex-end-19\\"><div class=\\"MuiGrid-item-2\\"><button tabindex=\\"0\\" class=\\"MuiButtonBase-root-179 MuiButton-root-206\\" type=\\"button\\" id=\\"removeAllButton\\"><span class=\\"MuiButton-label-207\\">Remove all ingredients</span><span class=\\"MuiTouchRipple-root-194\\"></span></button></div><div class=\\"MuiGrid-item-2\\"><button tabindex=\\"0\\" class=\\"MuiButtonBase-root-179 MuiButton-root-206 MuiButton-raised-212 MuiButton-raisedPrimary-213\\" type=\\"button\\" id=\\"saveButton\\"><span class=\\"MuiButton-label-207\\">Save recipe</span><span class=\\"MuiTouchRipple-root-194\\"></span></button></div></div></div>"`;
